src/lib/device/device_manager.cpp

```diff
@@ -204,11 +205,12 @@ shared< VirtualDev > DeviceManager::create_vdev(vdev_parameters&& vparam) {
         pdevs.erase(pdevs.begin() + 1, pdevs.end()); // TODO: Pick random one
     }
 
-    // Adjust the maximum number chunks requested before round up vdev size.
-    uint32_t max_num_chunks = 0;
-    for (const auto& d : m_dev_infos) {
-        max_num_chunks += hs_super_blk::max_chunks_in_pdev(d);
-    }
+    // Based on the min chunk size, we calculate the max number of chunks that can be created in each target pdev
+    uint32_t min_chunk_size = hs_super_blk::min_chunk_size(vparam.dev_type);
+    // FIXME: it is possible that each vdev is less than max_num_chunks, but total is more than MAX_CHUNKS_IN_SYSTEM.
+    uint32_t max_num_chunks = vparam.vdev_size / min_chunk_size > hs_super_blk::MAX_CHUNKS_IN_SYSTEM
```

Comment:
How about: `uint32_t max_num_chunks = std::min(vparam.vdev_size / min_chunk_size, hs_super_blk::MAX_CHUNKS_IN_SYSTEM);`

Reply:
the vparam.vdev_size / min_chunk_size is a uint64 so I tried to avoid the complexity of force type convert ---  i.e `vparam.vdev_size / min_chunk_size` can overflow uint32

Reply:
oh it is the same, sorry will fix that

Reply:
it is theoretically possible but not practically possible  as `min_chunk_size = 16MB * max_uint_32` = 2^56 = 64PB...

We will not have that big device. Will change to use force type convert.